src/prism.c

```diff
@@ -11906,8 +11906,19 @@ parse_block_parameters(
             false
         );
     }
-
     pm_block_parameters_node_t *block_parameters = pm_block_parameters_node_create(parser, parameters, opening);
+
+    if (
```

Comment:
It's probably totally fine to do it here, but I'm wondering if you could do it inside of `parse_parameters` instead and take a pointer to a boolean.

The reason is, this is going to check every parameter type every time, whereas if we do it in the `parse_parameters` loop, then it's only going to check it when parameters are actually added.

Could you try that approach?

Reply:
Sure I'll try. I've thought about it, but the checks here seemed quite cheap to me so that's why I implemented it in this way.

Reply:
Hmm that's a good point. I'll do some benchmarks and get back to you, hold off for now.